### PR TITLE
Add Processing Team Spec and Interfaces 

### LINKED ISSUE #19: processing-team-member-111701024-spec
Rakesh's (111701024) design specs should be added.

### PR DESCRIPTION
**Pull request by sakshi1603**
Closes #19 #18 #17 #15 #14 #13 #12 #10 #4

### CODE REVIEW

File: src/processing/IDrawShapes.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+*
+* @author Himanshu Jain
+*/
+
+public interface IDrawShapes {
+    // parameter: Pixel center - center position with intensity for circle
+    //            radius - radius of circle
+    void draw_circle(Pixel center, float radius);
```
Comment (convict-git) [inline src/processing/IDrawShapes.java]: [Coding convention for methods](https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names
)
Please refer to the coding conventions and do necessary changes.

File: src/processing/Processor.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		return;
+	}
+	
+	@Override
+	public void erase (ArrayList <Position> position) {
+		return;
+	}
+	
+	@Override
+	public void draw_circle(Pixel center, float radius) {
```
Comment (convict-git) [inline src/processing/Processor.java]: Please keep consistency in the coding standards. lower camel case is expected for method names.
[Method names convention](https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names)

File: src/processing/IUser.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+/**
+*
+* @author Himanshu Jain
+*/
+
+public interface IUser {
+    // this function is used to get the user details during initial setup from user through UI 
+    // if the board Id is null then we get a new board id from server and pass to UI
+    // parameter: userName - user name
```
Comment (convict-git) [inline src/processing/IUser.java]: Please refer to the Javadoc formatting guidelines. Kindly make use of block tags
[Java Formatting](https://google.github.io/styleguide/javaguide.html#s7-javadoc)

**Review by convict-git — CHANGES_REQUESTED** [review]
Kindly refer [Google Java Guidelines](https://google.github.io/styleguide/javaguide.html) and make necessary changes. Also reviewers at L2, please make a note of this.

**Comment by himanshu2047** [comment]
Ready for merge

**Comment by convict-git** [comment]
Amazing work! Thanks, guys. :+1:  Will see tomorrow morning and merge soon. 
@muhammedyaseen-git Kindly review once, it's pending right now from your side.

**Review by muhammedyaseen-git — APPROVED** [review]

**Review by convict-git — APPROVED** [review]

**Comment by convict-git** [comment]
Can you please change the Interfaces mentioned in the specs to follow the guidelines as you did for the actual interfaces? Most people will follow the documentation to understand the interface, and currently, there's a mismatch.

**Review by convict-git — APPROVED** [review]

**Pull request merged**